diff --git a/libs/frontend/uix/content/avatar.js b/libs/frontend/uix/content/avatar.js
index 38cf92c..a0d93a6 100644
--- a/libs/frontend/uix/content/avatar.js
+++ b/libs/frontend/uix/content/avatar.js
@@ -11,21 +11,21 @@ const Avatar = {
   theme: ({ DimensionSizes, commonStyles }) => ({
     "uix-avatar": { ...commonStyles, size: DimensionSizes },
     "uix-avatar__img": { _base: "", size: DimensionSizes },
-  }),
   render() {
     const { src, alt, placeholder } = this;
     let content;
     if (src) {
-      content = html`<img
+      content = html`<img 
         src=${src}
-        class=${this.theme("uix-avatar__img")}
+        data-theme="uix-avatar__img"
         alt=${alt}
       />`;
     } else if (placeholder) {
-      content = html`<span class=${this.theme("uix-avatar__img")}
+      content = html`<span data-theme="uix-avatar__img"
         >${placeholder}</span
       >`;
     }
+    }
 
     return html` <div class=${this.theme("uix-avatar")}>${content}</div> `;
   },
