--- libs/frontend/uix/content/avatar.js
+++ libs/frontend/uix/content/avatar.js
@@ -14,12 +14,12 @@
   render() {
     const { src, alt, placeholder } = this;
     let content;
     if (src) {
-      content = html`<img
+      content = html`<img 
         src=${src}
-        class=${this.theme("uix-avatar__img")}
+        data-theme="uix-avatar__img"
         alt=${alt}
       />`;
     } else if (placeholder) {
-      content = html`<span class=${this.theme("uix-avatar__img")}
+      content = html`<span data-theme="uix-avatar__img"
         >${placeholder}</span
       >`;
     }
--- libs/frontend/uix/content/icon.js
+++ libs/frontend/uix/content/icon.js
@@ -7,7 +7,7 @@
     size: T.string({ defaultValue: "" }),
     containerClass: T.string(),
   },
-  render() {
+  render() { 
     const { name } = this;
     return html`<ion-icon name=${name} role="img"></ion-icon>`;
   },
--- libs/frontend/uix/content/text.js
+++ libs/frontend/uix/content/text.js
@@ -38,7 +38,7 @@
     const tag = TAG_MAP[size] || "p";
 
     return staticHtml`
-      <${unsafeStatic(tag)} class="${unsafeStatic(
+      <${unsafeStatic(tag)} data-theme="${unsafeStatic(
       `${this.theme("uix-text__element")}`,
 )}">
         <slot></slot>
--- libs/frontend/uix/crud/crud-actions.js
+++ libs/frontend/uix/crud/crud-actions.js
@@ -23,7 +23,7 @@
       ></uix-crud-new-modal>
       <uix-button
         dropdown="hide"
-        class=${this.theme("uix-crud-actions__button")}
+        data-theme="uix-crud-actions__button"
       >
         <uix-icon name="chevron-down"></uix-icon> Actions
         <ul slot="dropdown" class=${this.theme("uix-crud-actions__dropdown")}>
--- libs/frontend/uix/crud/crud-detail.js
+++ libs/frontend/uix/crud/crud-detail.js
@@ -22,7 +22,7 @@
           </div>
         `,
   )}
-      <div class=${this.theme("uix-crud-detail__actions")}>
+      <div data-theme="uix-crud-detail__actions">
         <uix-button @click=${() => this.editItem(this.item)}> Edit </uix-button>
         <uix-button variant="error" @click=${() => this.deleteItem(this.item)}>
           Delete
--- libs/frontend/uix/crud/crud-new-modal.js
+++ libs/frontend/uix/crud/crud-new-modal.js
@@ -20,13 +20,13 @@
             <uix-icon-button
               slot="button"
               icon=${icon}
-              class=${this.theme("uix-crud-new-modal__icon")}
+              data-theme="uix-crud-new-modal__icon"
             ></uix-icon-button>`
           : html`<uix-button
               slot="button"
               variant="primary"
-              class=${this.theme("uix-crud-new-modal__button")}
+              data-theme="uix-crud-new-modal__button"
               >+ new</uix-button
             >`}
         <uix-form
--- libs/frontend/uix/crud/crud-search.js
+++ libs/frontend/uix/crud/crud-search.js
@@ -15,18 +15,18 @@
   },
   render() {
     return html`
-      <form class=${this.theme("uix-crud-search__form")}>
+      <form data-theme="uix-crud-search__form">
         <label for="simple-search" class="sr-only">Search</label>
-        <div class=${this.theme("uix-crud-search__input-container")}>
-          <div class=${this.theme("uix-crud-search__input-icon")}>
+        <div data-theme="uix-crud-search__input-container">
+          <div data-theme="uix-crud-search__input-icon">
             <uix-icon name="search"></uix-icon>
           </div>
           <input
             type="text"
             id="simple-search"
-            class=${this.theme("uix-crud-search__input")}
+            data-theme="uix-crud-search__input"
             placeholder="Search"
             required=""
           />
         </div>
--- libs/frontend/uix/crud/crud.js
+++ libs/frontend/uix/crud/crud.js
@@ -15,7 +15,7 @@
   },
   render() {
     return html`
-      <div class=${this.theme("uix-crud__actions")}>
+      <div data-theme="uix-crud__actions">
         <uix-crud-search
           .model=${this.model}
           .setRows=${this.setRows}
@@ -28,7 +28,7 @@
           .ModelClass=${this.ModelClass}
         ></uix-crud-actions>
       </div>
-      <div class=${this.theme("uix-crud__table")}>
+      <div data-theme="uix-crud__table">
         <uix-crud-table
           .rows=${this.rows}
           .fields=${this.fields}